### PR TITLE
Ticket5333 helium recovery fins plc

### PR DESCRIPTION
**Pull request by Alexandru-Dascalu**
Adds IOC Tests for the FINS IOC specific to the Helium Gas Recovery PLC. Part of https://github.com/ISISComputingGroup/IBEX/issues/5333 .

### CODE REVIEW

File: tests/helium_recovery_plc.py [hunk 1]
```diff
@@ -0,0 +1,416 @@
...
+
+    @parameterized.expand(parameterized_list(COLDBOX_STATUS_VALUES))
+    @skip_if_devsim("sim pvs not available in devsim")
+    def test_WHEN_coldbox_status_set_sim_pv_THEN_ioc_read_correctly(self, _, test_value):
+        index_test_value = COLDBOX_STATUS_VALUES.index(test_value)
+        self.ca.set_pv_value("SIM:COLDBOX:STATUS", index_test_value)
+        self.ca.assert_that_pv_after_processing_is("COLDBOX:STATUS", test_value)
+
+    @parameterized.expand(parameterized_list(itertools.product(VALVE_STATUS_PVS, VALVE_STATUS_VALUES)))
+    @skip_if_recsim("lewis backddor not available in recsim")
```
Comment (JamesKingWork) [inline tests/helium_recovery_plc.py:330]: typo backdoor

File: tests/helium_recovery_plc.py [hunk 2]
```diff
@@ -0,0 +1,416 @@
...
+    # mbbiDirect recors work properly.
+
+    @parameterized.expand(parameterized_list(LIQUEFIER_ALARMS))
+    @skip_if_recsim("lewis backdoor not available in recsim")
+    def test_WHEN_liquefier_alarm_set_backdoor_THEN_ioc_read_correctly(self, _, pv_name):
+        alarm_index = LIQUEFIER_ALARMS.index(pv_name)
+        mbbi_direct_pv = HeliumRecoveryPLCTests._get_liquefier_hardware_pv(alarm_index)
+        test_value = HeliumRecoveryPLCTests._get_alarm_test_value(mbbi_direct_pv, alarm_index)
+
+        self.ca.process_pv("{}:_RAW".format(mbbi_direct_pv))
```
Comment (JamesKingWork) [inline tests/helium_recovery_plc.py:356]: This string and others are formatted 3 times separately, move out into a variable

Comment (Alexandru-Dascalu) [reply tests/helium_recovery_plc.py]: Done.

File: tests/helium_recovery_plc.py (hunk 1 above)
Comment (Alexandru-Dascalu) [reply tests/helium_recovery_plc.py]: Done.

File: tests/helium_recovery_plc.py [hunk 3]
```diff
@@ -0,0 +1,416 @@
...
+        self.ca.assert_that_pv_after_processing_is(pv_name, "MANUAL")
+
+        self._lewis.backdoor_run_function_on_device("set_memory", (pv_name, 2))
+        self.ca.assert_that_pv_after_processing_is(pv_name, "AUTOMATIC")
+
+        self._lewis.backdoor_run_function_on_device("set_memory", (pv_name, 1))
+        self.ca.assert_that_pv_after_processing_is(pv_name, "MANUAL")
+
+    @parameterized.expand(parameterized_list(AUTO_MANUAL_PV_NAMES))
+    @skip_if_devsim("sim pvs not available in recsim")
```
[the remark this answers is not in the document]
Comment (Alexandru-Dascalu) [reply tests/helium_recovery_plc.py]: Done, thanks for spotting it

File: tests/helium_recovery_plc.py [hunk 4]
```diff
@@ -0,0 +1,416 @@
...
+        self.ca.set_pv_value("SIM:{}".format(mbbi_direct_pv), 0)
+        self.ca.process_pv("{}:_RAW".format(mbbi_direct_pv))
+        self.ca.assert_that_pv_is("{}:ALARM".format(pv_name), "OK")
+
+    @staticmethod
+    def _get_liquefier_hardware_pv(alarm_index):
+        """
+        The 24 alarm bi records get their value from one of two mbbi records. The first 15 get their value from the
+        first, and the rest from the second one.
+        :param alarm_index: Index of the pv name in the list of alarm PVs.
```
[the remark this answers is not in the document]
Comment (Alexandru-Dascalu) [reply tests/helium_recovery_plc.py]: Done.

**Pull request merged**